examples/model_compress/pruning/mobilenet_end2end/pretrain.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from time import gmtime, strftime
+import torch
+import torch.nn as nn
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+import numpy as np
+
+from utils import *
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+model_type = 'mobilenet_v2_torchhub'   # 'mobilenet_v1' 'mobilenet_v2' 'mobilenet_v2_torchhub'
+pretrained = True                     # load imagenet weight (only for 'mobilenet_v2_torchhub')
+experiment_dir = 'pretrained_{}'.format(model_type)
+os.mkdir(experiment_dir)
+checkpoint = None
+input_size = 224
+n_classes = 120
+
+# optimization parameters
+batch_size = 32
+n_epochs = 160
+learning_rate = 1e-4         # 1e-4 for finetuning, 1e-3 (?) for training from scratch
+weight_decay = 0.0   #1e-3          # l2 regularization
+
```

Comment:
maybe these need an arg parser? or the user must modify the source code to change the model.

Reply:
fixed